idp-service/src/main/java/io/mosip/idp/services/OAuthServiceImpl.java

```diff
@@ -86,19 +83,24 @@ public TokenResponse getTokens(@Valid TokenRequest tokenRequest) throws IdPExcep
         kycExchangeRequest.setClientId(tokenRequest.getClient_id());
         kycExchangeRequest.setKycToken(transaction.getKycToken());
         kycExchangeRequest.setAcceptedClaims(transaction.getAcceptedClaims());
-        String encryptedKyc = authenticationWrapper.doKycExchange(kycExchangeRequest);
+        kycExchangeRequest.setClaimsLocales(transaction.getClaimsLocales());
+        ResponseWrapper<KycExchangeResult> exchangeResult = authenticationWrapper.doKycExchange(kycExchangeRequest);
+
+        if(exchangeResult.getErrors() != null && !exchangeResult.getErrors().isEmpty()) {
+            throw new IdPException(exchangeResult.getErrors().get(0).getErrorCode());
+        }
 
         TokenResponse tokenResponse = new TokenResponse();
         tokenResponse.setAccess_token(tokenService.getAccessToken(transaction));
         String accessTokenHash = IdentityProviderUtil.generateAccessTokenHash(tokenResponse.getAccess_token());
         transaction.setAHash(accessTokenHash);
         tokenResponse.setId_token(tokenService.getIDToken(transaction));
         tokenResponse.setExpires_in(accessTokenExpireSeconds);
-        tokenResponse.setScope(transaction.getScopes());
+        tokenResponse.setToken_type(Constants.BEARER);
 
         // cache kyc with access-token as key
         transaction.setIdHash(IdentityProviderUtil.generateAccessTokenHash(tokenResponse.getId_token()));
```

Comment:
should we rename this method to just generateTokenHash, so it is ok for us to use for both access token and id token

Reply:
I have created a separate method because 
"hash the access_token value with SHA-256, then take the left-most 128 bits and base64url encode them. The at_hash value is a case-sensitive string" is mentioned in SPEC.

Reply:
ok, but my concern is the method is called generateAccessTokenHash but we are passing the id token here, so was asking if we can give a name which not confusing